harfbuzz/src/blob.rs

```diff
@@ -7,18 +7,34 @@
 // option. This file may not be copied, modified, or distributed
 // except according to those terms.
 
-use std::{mem, ops, ptr, slice};
+use std::marker::PhantomData;
 use std::os::raw::{c_char, c_uint};
+use std::{mem, ops, ptr, slice};
 use sys;
 
-/// Wrapper type that owns an `hb_blob_t`.
-pub struct Blob {
+/// Blobs wrap a chunk of binary data to handle lifecycle management of data
+/// while it is passed between client and HarfBuzz.
+///
+/// Blobs are primarily used to create font faces, but also to access font face
+/// tables, as well as pass around other binary data.
+pub struct Blob<'a> {
     raw: *mut sys::hb_blob_t,
+    phantom: PhantomData<&'a sys::hb_blob_t>,
```

Comment:
Just as a style point, I'd have the phantom refer to `&'a [u8]`, but functionally it makes no difference.

Reply:
Done!